ui.frontend/src/pages/HistoryPage.tsx

```diff
@@ -9,29 +9,55 @@ import Pause from '@spectrum-icons/workflow/Pause';
 import Search from '@spectrum-icons/workflow/Search';
 import Star from '@spectrum-icons/workflow/Star';
 import { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useSearchParams } from 'react-router-dom';
 import { useDebounce } from 'react-use';
 import DateExplained from '../components/DateExplained';
 import ExecutableValue from '../components/ExecutableValue.tsx';
 import ExecutionStatusBadge from '../components/ExecutionStatusBadge.tsx';
+import { searchParams } from '../constants/searchParams.ts';
+import { useFormatter } from '../hooks/formatter';
 import { toastRequest } from '../utils/api';
 import { ExecutionOutput, ExecutionStatus } from '../utils/api.types';
-import { useFormatter } from '../hooks/formatter';
+
+const getDateFromStringValue = (urlValue: string | null, fallback?: CalendarDateTime) => {
+  if (urlValue) {
+    const date = new Date(urlValue);
+
+    return new CalendarDateTime(date.getFullYear(), date.getMonth() + 1, date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds());
+  }
+
+  return fallback ?? null;
+};
 
 const HistoryPage = () => {
   const navigate = useNavigate();
   const [executions, setExecutions] = useState<ExecutionOutput | null>(null);
   const [loading, setLoading] = useState<boolean>(false);
 
-  const now = new Date(); // assume start date to be 7 days ago
-  const startDateDefault = new CalendarDateTime(now.getFullYear(), now.getMonth() + 1, now.getDate() - 7, 0, 0, 0);
+  const sevenDaysAgo = new Date(); // assume start date to be 7 days ago
+  sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);
+  const startDateDefault = new CalendarDateTime(sevenDaysAgo.getFullYear(), sevenDaysAgo.getMonth() + 1, sevenDaysAgo.getDate(), 0, 0, 0);
+
+  const [searchState, setSearchState] = useSearchParams();
+  const [startDate, setStartDate] = useState<DateValue | null>(getDateFromStringValue(searchState.get(searchParams.startDate), startDateDefault));
+  const [endDate, setEndDate] = useState<DateValue | null>(getDateFromStringValue(searchState.get(searchParams.endDate)));
+  const [status, setStatus] = useState<string | null>(searchState.get(searchParams.status) || 'all');
+  const [executableId, setExecutableId] = useState<string>(searchState.get(searchParams.executableId) || '');
+
+  const [durationMinFromParam, durationMaxFromParam] = (() => {
```

Comment:
I had some problems with that approach; that's why I split duration to separate fields; are you really need to have it together?

Reply:
i just moved extracting from url to one function, formula is not changed. Anyway, everything seems to be working fine, so honeslty don't see anything dangerous here

Reply:
And if it is how backend need it, I would stay consistent

Reply:
hmm IMO previous version required less code ; and it was more readable